### PR TITLE
tests: add UI test suite

### PR DESCRIPTION
**Pull request by jlejoux**
This adds some basic 3DS1/2 UI tests:

- Card tokenization
- Alternative payment listing
- Successful 3DS payment (web)
- Successful 3DS2 payment (native)
- Failed 3DS1 payment (web)
- Failed 3DS2 challenge (native)
- Successful makeCardToken (native)
- Failed makeCardToken (web)
- Failed makeCardToken challenge (native)

### CODE REVIEW

File: processout-sdk/src/androidTest/java/com/processout/processout_sdk/UITestSuite.java [hunk 1]
```diff
@@ -0,0 +1,664 @@
...
+                        Invoice invoiceResult = null;
+                        try {
+                            invoiceResult = gson.fromJson(json.getJSONObject("invoice").toString(), Invoice.class);
+                        } catch (JSONException e) {
+                            fail("Unhandled exception");
+                            e.printStackTrace();
+                            return;
+                        }
+
+                        p.makeCardPayment(invoiceResult.getId(), token, new ThreeDSHandler() {
```
[the remark this answers is not in the document]
Comment (jlejoux) [reply processout-sdk/src/androidTest/java/com/processout/processout_sdk/UITestSuite.java]: - I'll make changes to split the tests 👍 
- The CI might be doable but a bit complicated as we have to run an android emulator to run the tests 🤔

**Comment by asiragusa** [comment]
@manuel-huez I will plug automatic tests when we'll finish with the new CI platform

**Comment by asiragusa** [comment]
also @manuel-huez you can't break a Java class in multiple files and generally it's not a good idea. The best way to avoid files that are thousands lines long is to refactor the code. In any case it's not shocking to have a java class with ~500 lines of code for testing

**Comment by jlejoux** [comment]
I can split tests into different categories but it forces me to create some kind of static class that has different parameters required for the tests. It adds a bit of complexity but can be done I guess.

**Pull request merged**